### PR TITLE
ENH: add GWGradientBoostingRegressor

### PR DESCRIPTION
**Pull request by FirePheonix**
Added Gradient Boosting Regressor - implemented the algorithm, it's test and also added in ipynb notebook.
For issue: #48

### CODE REVIEW

**Comment by FirePheonix** [comment]
hey @martinfleis so i tested it out on local - it highly overfits on ```include_focal = true``` , and gives R^2 values as ~0.9999999 - but the values on ```include_focal = false``` it performs poorly on the training set and shows some bad, but honest numbers.

image

And also, for the example on Random Forest Regressor - the values are also overfit and all of them give ~0.8 since it ```includes_focal = true``` by default
image

For now, I've added the actual values in this commit and testing if CI works. I'll change immediately if you say so.

**Comment by martinfleis** [comment]
We don't have to show local r2 but can show pred_ instead. In any case, none of the ensemble models should include focal by default. I did set it to False for ensemble classifiers but forgot in RF regressor. Can you update that as well?

File: gwlearn/ensemble.py
Comment (martinfleis) [inline gwlearn/ensemble.py]:
```suggestion
        False
```

File: gwlearn/ensemble.py [hunk 1]
```diff
@@ -784,3 +785,222 @@ def fit(
...
+    >>> gdf = gpd.read_file(get_path('geoda.guerry'))
+    >>> X = gdf[['Crm_prp', 'Litercy', 'Donatns', 'Lottery']]
+    >>> y = gdf["Suicids"]
+
+    >>> gw = GWGradientBoostingRegressor(
+    ...     bandwidth=30,
+    ...     fixed=False,
+    ...     random_state=0,
+    ... ).fit(X, y, geometry=gdf.representative_point())
+    >>> gw.local_r2_.head()
```
Comment (martinfleis) [inline gwlearn/ensemble.py]: Again, can we replace this with `pred_`?

**Review by martinfleis — COMMENTED** [review]
Cool! A few notes.

**Comment by FirePheonix** [comment]
@martinfleis shall i make the pred_ changes inside the ensemble.ipynb file too and run again?
